src/app/src/components/Submissions/Detail/Map.js

```diff
@@ -128,31 +137,75 @@ function MapButton({ icon, children, onClick }) {
     );
 }
 
-function SubmissionStatusBar({ status }) {
+function useSubmissionStatusBar(boundary) {
+    const userRole = useSelector(state => state.auth.user.role);
+
+    if (userRole === ROLES.VALIDATOR) {
+        return null;
+    }
+
+    if (userRole === ROLES.VALIDATOR) {
+        return null;
+    }
+
     const getDetailText = () => {
-        switch (status) {
-            case BOUNDARY_STATUS.DRAFT:
-                return 'This map is a draft.';
+        switch (boundary.status) {
+            case BOUNDARY_STATUS.SUBMITTED:
+            case BOUNDARY_STATUS.IN_REVIEW:
+                return 'Your map will be reviewed.';
+            case BOUNDARY_STATUS.NEEDS_REVISIONS:
+                return getReviewDetailText(boundary.submission.review);
+            case BOUNDARY_STATUS.APPROVED:
+                return 'Your map has been approved.';
+            default:
+                return null;
+        }
+    };
+
+    const getReviewDetailText = review => {
+        const commentCount = review.comments.length;
+
+        if (commentCount === 1) {
+            return 'There is one comment on your map.';
+        }
+
+        if (commentCount > 1) {
+            return `There are ${commentCount} comments on your map.`;
+        }
+
+        return 'Your map have been reviewed and needs revisions.';
+    };
+
+    const getBarColor = () => {
+        switch (boundary.status) {
             case BOUNDARY_STATUS.SUBMITTED:
+                return 'teal';
             case BOUNDARY_STATUS.IN_REVIEW:
-                return 'This map will be reviewed.';
             case BOUNDARY_STATUS.NEEDS_REVISIONS:
-                return 'This map needs revisions.';
+                return 'orange';
             case BOUNDARY_STATUS.APPROVED:
-                return 'This map has been approved.';
+                return 'green';
             default:
-                throw new Error(`Unexpected status ${status}`);
+                return null;
         }
     };
```

Comment:
This diff is confusing. `getBarColor` looks like this
```js

    const getBarColor = () => {
        switch (boundary.status) {
            case BOUNDARY_STATUS.SUBMITTED:
                return 'teal';
            case BOUNDARY_STATUS.IN_REVIEW:
            case BOUNDARY_STATUS.NEEDS_REVISIONS:
                return 'orange';
            case BOUNDARY_STATUS.APPROVED:
                return 'green';
            default:
                return null;
        }
    };
```
This leads to a mismatch between the badge colors and the submission bar color. `SUBMITTED` and `IN_REVIEW` have the same status bar text, so I figured they should have the same color as well.

I assumed `APPROVED` should be green, but maybe we don't need a status bar when a boundary is approved.

Reply:
> maybe we don't need a status bar when a boundary is approved.

Correct on the green. There is a bar for approved boundaries, shown here:

![image](https://user-images.githubusercontent.com/1430060/197589029-3b7e5f65-fa6d-40c3-a658-70466bca47b9.png)